lib/fprint.c

```diff
@@ -459,11 +450,14 @@ void fpCachePopulate(fingerPrintCache fpc, rpmts ts, int fileCount)
 
     rpmFpHash symlinks = rpmFpHashCreate(fileCount/16+16, fpHashFunction, fpEqual, NULL, NULL);
 
+    /* create a hash of all symlinks in the new packages */
     pi = rpmtsiInit(ts);
     while ((p = rpmtsiNext(pi, 0)) != NULL) {
 	fingerPrint *fpList;
 	(void) rpmsqPoll();
 
+	if (rpmteType(p) == TR_REMOVED)
+	    continue;		/* we are only interested in new packages */
```

Comment:
Why not change the loop itself to only look at added packages instead, eg rpmtsiNext(pi, TR_ADDED) in the above?

Reply:
Oh, I didn't know about that. Force pushed.